src/overlays/actors/ovl_Bg_Ingate/z_bg_ingate.c

```diff
@@ -43,30 +43,32 @@ ActorInit Bg_Ingate_InitVars = {
  *
  * @param this
  * @param play
- * @param actorCat - Category of Actor
+ * @param actorCategory - Category of Actor
  * @param actorId - ID of actor to search for
  * @return Actor*
  */
-Actor* BgIngate_FindActor(BgIngate* this, PlayState* play, u8 actorCat, s16 actorId) {
-    Actor* foundActor = NULL;
-    Actor* tempActor;
+Actor* BgIngate_FindActor(BgIngate* this, PlayState* play, u8 actorCategory, s16 actorId) {
+    Actor* actorIter = NULL;
 
     while (true) {
-        foundActor = SubS_FindActor(play, foundActor, actorCat, actorId);
+        actorIter = SubS_FindActor(play, actorIter, actorCategory, actorId);
 
-        if ((foundActor == NULL) || (((this != (BgIngate*)foundActor)) && (foundActor->update != NULL))) {
+        if (actorIter == NULL) {
             break;
         }
 
-        tempActor = foundActor->next;
-        if (tempActor == NULL) {
-            foundActor = NULL;
+        if ((this != (BgIngate*)actorIter) && (actorIter->update != NULL)) {
```

Comment:
Why split the condition chain into two? Maybe the diff is just making it look weird?